File: Python/refer-offer/diao_zheng_shu_zu_shun_xu_shi_qi_shu_wei_yu_ou_shu_qian_mian_lcof.py
from typing import *

def exchange(nums: List[int]) -> List[int]:
    left, right = 0, len(nums) - 1
    while left < right:
        while left < right and nums[left] % 2 == 1:
            left += 1
        while left < right and nums[right] % 2 == 0:
            right -= 1
        
        if left < right:
            temp = nums[left]
            nums[left] = nums[right]
            nums[right] = temp
        
        left += 1
        right -=1
    
    return nums

File: Java/src/com/dbc/algorithm/ReferOffer/DiaoZhengShuZuShunXuShiQiShuWeiYuOuShuQianMianLcof.java
package com.dbc.algorithm.ReferOffer;

public class DiaoZhengShuZuShunXuShiQiShuWeiYuOuShuQianMianLcof {
    public int[] exchange(int[] nums) {
        int left = 0, right = nums.length - 1;
        while (left < right) {
            while (left < right && nums[left] % 2 == 1) {
                left++;
            }
            while (left < right && nums[right] % 2 == 0) {
                right--;
            }
            if (left < right) {
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
            }
        }
        return nums;
    }
}
